lib/split/trial.rb

```diff
@@ -93,12 +95,14 @@ def choose!(context = nil)
 
       new_participant_and_cohorting_disabled = new_participant && @experiment.cohorting_disabled?
 
-      @user[@experiment.key] = alternative.name unless @experiment.has_winner? || !should_store_alternative? || new_participant_and_cohorting_disabled
+      @user[user_experiment_key] = alternative.name unless @experiment.has_winner? || !should_store_alternative? || new_participant_and_cohorting_disabled
       @alternative_choosen = true
       run_callback context, Split.configuration.on_trial unless @options[:disabled] || Split.configuration.disabled? || new_participant_and_cohorting_disabled
       alternative
     end
 
+    private
```

Comment:
Moved the private block up as `within_conversion_time_frame? `is used only internally and should have been made private. The spec coverage only covers its usage through `complete!`